src/components/main-panel/main-panel.tsx

```diff
@@ -108,38 +116,107 @@ export const MainPanel = () => {
         }
     };
 
-    const handleDragEnd = (event: DragEndEvent) => {
-        const { active, over } = event;
+    const handleOnDragStart = (event: DragStartEvent) => {
+        dragStartY.current = event.active.rect.current.initial?.top;
+        elementRefs.current.forEach((elementRef, index) => {
+            elementStartingRefs.current[index] =
+                elementRef.element.getBoundingClientRect();
+        });
+    };
 
-        if (active.id !== over.id) {
-            const oldIndex = orderedIdentifiers.findIndex(
-                (item) => item.internalId === active.id
-            );
-            const newIndex = orderedIdentifiers.findIndex(
-                (item) => item.internalId === over.id
+    const handleOnDragMove = debounce((event: DragMoveEvent) => {
+        const { active } = event;
+
+        const wrapperBounds = wrapperRef.current?.getBoundingClientRect();
+
+        const wrapperLeftThirds = wrapperBounds.left + wrapperBounds.width / 3;
+        const wrapperRightThirds =
+            wrapperBounds.left +
+            (wrapperBounds.width * 2) / 3 -
+            wrapperBounds.width / 9;
+
+        const currentActiveIndex = orderedIdentifiers.findIndex(
+            (item) => item.internalId === active.id
+        );
+
+        const mouseX =
+            active.rect.current.translated?.left +
+            event.activatorEvent["layerX"];
+
+        let elementIndex = currentActiveIndex;
+        let column: IElementIdentifier["size"] = "full";
```

Comment:
let's use `size` since the rest of the logic is using that
```suggestion
        let size: IElementIdentifier["size"] = "full";
```